docs/usage/edsl_examples.rst

```diff
@@ -314,15 +326,270 @@ coefficients, and ``P`` gives the padding offsets.
+    {\color{purple}\verb!y!}
+
+.. tabs::
+
+  .. group-tab:: C++
+
+    .. literalinclude:: ../../plaidml/edsl/examples/gemv.cc
+      :language: cpp
+      :start-after: gemv_start
+      :end-before: gemv_end
+
+  .. group-tab:: Python
+
+    .. literalinclude:: ../../plaidml/edsl/examples/gemv.py
+      :pyobject: gemv
+
+Constant-vector multiplication and constant-tensor multiplication can be handled using the 
+element-wise :math:`{\color{magenta}\verb!*!}` operator in eDSL. Thus :math:`\color{turquoise}\alpha` and  
+:math:`\color{turquoise}\beta` can handled as follows. 
+
+.. math::
+
+  \begin{aligned}
+  &\color{blue}\verb!A!
+  {\color{magenta}\verb!  *  !}
+  \color{turquoise}\verb!alpha!\\
+  &\color{blue}\verb!y!
+  {\color{magenta}\verb!  *  !}
+  \color{turquoise}\verb!beta!
+  \end{aligned}
+
+.. tabs::
+
+  .. group-tab:: C++
+
+    .. literalinclude:: ../../plaidml/edsl/examples/gemv.cc
+      :language: cpp
+      :start-after: constant_gemv_start
+      :end-before: constant_gemv_end
+
+  .. group-tab:: Python
+
+    .. literalinclude:: ../../plaidml/edsl/examples/gemv.py
+      :pyobject: gemv2
+
+GEMM BLAS Level 3
+*******************
+This example illustrates matrix-matrix operations using the generalized matrix-matrix (GEMM) multiplication of the form:
+
+
+.. math::
+
+   \color{red}O
+   \color{default}  =
+   \color{turquoise} \alpha
+   \color{blue}A
+   \color{orange} *
+   \color{blue} B
+   \color{green} +
+   \color{turquoise} \beta
+   \color{blue}C
+
+
+Here :math:`\color{blue}A` is a matrix, :math:`\color{purple}x` and :math:`\color{purple}y` 
+are vectors and :math:`\color{turquoise}\alpha` and  :math:`\color{turquoise}\beta` are 
+constants. 
+
+We can represent the matrix operation involved as:
+
+.. math::
+   {\color{red}O[i, j]}
+   \color{default}  =
+   \color{green} \sum_{k} (
+   \color{turquoise} \alpha
+   \color{magenta}\cdot
+   \color{blue}A[
+   \color{default}i, k
+   \color{blue}]
+   \color{orange} *
+   \color{blue} B[
+   \color{default}k, j
+   \color{blue}] 
+   \color{green} )
+   \color{magenta} +
+   \color{turquoise} \beta
+   \color{magenta}\cdot
+   \color{blue}C[
+   \color{default}i, j
+   \color{blue}]
+
+
+This can easily be written in eDSL as follows. 
+
+
+.. math::
+
+    \begin{aligned}
+    &\color{blue}\verb! A!
+    \color{default}\verb! =!
+    \color{blue}\verb! A!
+    \color{magenta}\verb! *!
+    \color{turquoise}\verb! alpha! \\
+    &\color{blue}\verb! C!
+    \color{default}\verb! =!
+    \color{blue}\verb! C!
+    \color{magenta}\verb! *!
+    \color{turquoise}\verb! beta! \\
+    \verb!Contraction()!
+    &\verb!.outShape(I,J)!\\
+    &{\color{red}\verb!.outAccess(i,j)!}\\
+    &{\color{green}\verb!.sum(!}\\
+    &{\color{blue}\verb!        A(!}
+    \verb!i,k!
+    {\color{blue}\verb!)!}
+    {\color{orange}\verb!  *  !}
+    {\color{blue}\verb!B(!}
+    \verb!k,j!
+    {\color{blue}\verb!)!}\\
+    &{\color{green}\verb!     )!}\\
```

Comment:
To my eye, at least, the interior of the sum and the final close `)` are indented ~1 tab too far. Perhaps remove the leading spaces?